src/pages/SignupPage/SignupPage.jsx

```diff
@@ -0,0 +1,159 @@
+import { Link, Navigate, useNavigate } from "react-router-dom";
+import { useState } from "react";
+import styles from "./SignupPage.module.css";
+import { validateInputs } from "../../utils";
+import axios from "axios";
+import { useUserDetails } from "../../contexts/UserContext/UserContext";
+
+const SignupPage = () => {
+  const [signupData, setSignupData] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    password: "",
+  });
+  const [confirmPass, setConfirmPass] = useState("");
+  const { dispatchUser } = useUserDetails();
+  const navigate = useNavigate();
+  const [errorMsg, setErrorMsg] = useState("");
+  console.log(signupData);
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setSignupData({ ...signupData, [name]: value });
+  };
+
+  const matchPasswords = (e) => {
+    if (e.target.value !== signupData.password) {
+      setErrorMsg("Passwords don't match");
+    } else {
+      setErrorMsg("");
+    }
+  };
+
+  const handleSignup = async (e) => {
+    e.preventDefault();
+    setErrorMsg("");
+    if (validateInputs(signupData.email, signupData.password)) {
+      try {
+        const response = await axios.post("/api/auth/signup", signupData);
+        if (response.status === 200 || response.status === 201) {
+          const { createdUser, encodedToken } = response.data;
+          localStorage.setItem("authToken",encodedToken)
+          dispatchUser({ type: "USER_LOGIN", payload: createdUser });
+          navigate("/explore/All");
+        }
+      } catch (err) {
+        setErrorMsg("Couldn't signup");
+        console.log("Uh!oh. Something went wrong", err);
+      }
+    } else {
+      setErrorMsg("Check email/password");
+    }
+  };
+
+  return (
+    <div className={`${styles.signUpPage} dark-theme generic-page`}>
+      <section className={`${styles.signupContainer} flex-vertical`}>
+        <h1 className={`heading1 brand-color gap-d30`}>Signup</h1>
+        <form className={`${styles.signupForm} flex-vertical`}>
+          <div className={`${styles.nameContainer} flex`}>
+            <div className={`input-wrapper gap-d20 ${styles.inputField}`}>
+              <label className="input-name txt-bold" for="input">
+                First Name
+              </label>
+              <input
+                className="input-box"
+                name="firstName"
+                type="text"
+                value={signupData.firstName}
+                placeholder="Enter first name"
+                required
+                autocomplete="on"
+                onChange={(e) => handleChange(e)}
+              />
+            </div>
+            <div className={`input-wrapper gap-d20 ${styles.inputField}`}>
+              <label className="input-name txt-bold" for="input">
```

Comment:
use htmlFor here as react ignores for

Reply:
Oh yes. Almost forgot this. Thanks for pointing it out.